precompiles/ArbOwner.go

```diff
@@ -57,6 +61,46 @@ func (con ArbOwner) GetAllChainOwners(c ctx, evm mech) ([]common.Address, error)
 	return c.State.ChainOwners().AllMembers(65536)
 }
 
+// AddChainOwner adds account as a native token owner
+func (con ArbOwner) AddNativeTokenOwner(c ctx, evm mech, newOwner addr) error {
+	if c.State.ArbOSVersion() < params.ArbosVersion_41 {
```

Comment:
I think we should use `PrecompileMethod.arbosVersion` field to do the version feature gating instead of returning error here. `arbosVersion` is checked during each call to a precompile method and returns similar output as if the method didn't exist:
https://github.com/OffchainLabs/nitro/blob/c8186eabd6b02f1c70aade188350583c44493b4c/precompiles/precompile.go#L686-L690

similarly to some other ArbOwner methods:
https://github.com/OffchainLabs/nitro/blob/c8186eabd6b02f1c70aade188350583c44493b4c/precompiles/precompile.go#L607-L612

Reply:
I believe that returning the same error here is important to avoid divergency of a precompile method execution  before arbos upgrade on the previous node software version versus execution still before arbos upgrade but with new node software version.